src/Internal/JsWebSocket.js

```diff
@@ -45,7 +45,7 @@ exports._onWsConnect = ws => fn => () => {
 
 exports._onWsError = ws => fn => () => {
   const listener = function (event) {
-    fn(event.toString())();
+    fn(event.message)();
```

Comment:
This code assumes that the thing being thrown is indeed an `Error`, but in fact any object can be thrown from JS, hence the `toString`. Note that when a message property is set properly in a class that extends Error, it gets propagated into a toString method:


```
> class A extends Error {
... constructor() {
..... super();
..... this.message = "hiii"
..... }
... }
undefined
> throw new A()
Thrown:
{ [Error: hiii] message: 'hiii' }
```

We could accept this change, but with one addition: we should use `message` only for `event instanceof Error` and only if it is a string (`typeof event.message === 'string'`), and `toString` for everything else.

Reply:
It's OK to specialize it for that error layout with `EventError.error`, but we also need a fallback case and to detect the error type correctly.